3rdparty/mbedtls/CMakeLists.txt

```diff
@@ -155,39 +155,42 @@ maybe_build_using_clangw(mbedtls)
 
 # Change the mbedcrypto_static target to the real mbedcrypto library name.
 set_target_properties(mbedcrypto_static PROPERTIES OUTPUT_NAME "mbedcrypto")
+if (LVI_MITIGATION MATCHES ControlFlow)
+  set_target_properties(mbedcrypto_static-lvi-cfg PROPERTIES OUTPUT_NAME "mbedcrypto-lvi-cfg")
```

Comment:
Does this work with the inline assembly -- e.g. https://github.com/openenclave/openenclave/blob/master/3rdparty/mbedtls/mbedtls/library/aesni.c#L125 ?

Reply:
Yes, this works with inline assembly. The instrumentation is done by the assembler.

Reply:
To be more precise, the mitigation that this PR enables supports inline assembly. However, the PR only supports the type of "Control-Flow-Mitigation" that the Intel's deep dive article [1] describles. That is to say, the supported mitigation does not take care of all the load instructions such as the one in your example. Instead, it focuses on control-flow pattern such as indirect branches and return. Please let me know you have more question.

[1] https://software.intel.com/security-software-guidance/insights/deep-dive-load-value-injection#applyenclavemitigation

Reply:
@mingweishih Will some of the cryptographic code need additional manual patching? The second case study in https://lviattack.eu/lvi.pdf has a similar example, but takes some extra assumptions for demonstration (known ciphertext, side-channel...)

Reply:
We currently don't have plan beyond the support of control-flow mitigation given that the cost of full mitigation is too high. If there is concern that control-flow mitigation is not sufficient, we will also consider adding further support.